terraform/environments/electronic-monitoring-data/glue-job/parquet_to_csv.py

```diff
@@ -0,0 +1,80 @@
+import sys
+from awsglue.transforms import *
+from awsglue.utils import getResolvedOptions
+from pyspark.context import SparkContext
+from awsglue.context import GlueContext
+from awsglue.job import Job
+
+# Set up Glue context and job
+args = getResolvedOptions(sys.argv, ["JOB_NAME"])
+sc = SparkContext()
+glueContext = GlueContext(sc)
+spark = glueContext.spark_session
+job = Job(glueContext)
+job.init(args["JOB_NAME"], args)
+
+# Source and destination buckets
+source_bucket = "source-bucket-name"
+destination_bucket = "destination-bucket-name"
+source_path = "s3://{}/".format(source_bucket)
+destination_path = "s3://{}/".format(destination_bucket)
+
+# Maximum size for each CSV file (in bytes)
+max_csv_size = 5 * 1024 * 1024 * 1024  # 5GB
+
+
+# Function to write DataFrame to CSV with size check and splitting
+def write_dataframe_to_csv(dataframe, destination_path):
+    # Check the size of the DataFrame
+    dataframe_size = dataframe.rdd.map(lambda x: len(str(x))).reduce(lambda x, y: x + y)
+
+    # If the size exceeds the maximum CSV size, split it into smaller chunks
+    if dataframe_size > max_csv_size:
+        # Split the DataFrame into smaller chunks
+        num_chunks = dataframe.rdd.getNumPartitions()
```

Comment:
Also not sure the number of partitions in the dataframe will be sensible so this might give you a pretty low number.

I wonder if maybe it's something more low tech like

```
num_rows = dataframe.count()
num_columns = len(dataframe.columns)

chunk_size = 100000  # no idea what is sensible

num_chunks = num_rows * num_columns / chunk_size

smaller_dataframes = dataframe.randomSplit(num_chunks)
```


Also just noticed that `dataframe.randomSplit([1.0 / num_chunks] * num_chunks)` will split into 1 chunk.